81 feature frontend for engine configuration group5

tr0ub1eM4k3r opened the pull request:
Closes #81 

### Changes/Additions:

- Added UI for the Engine definitions
- Added Modal component
- Moved validation helper functions to their own class

DenFlyvendeGed committed:
class and fields defined

tr0ub1eM4k3r committed:
Made general Dialog component

Co-authored-by: SolarEarth37 <SolarEarth37@users.noreply.github.com>

DenFlyvendeGed committed:
created EngineStore

SolarEarth37 committed:
Made EngineSeperate component
This allows for multiple engines to be shown in modal component
(Not tested, because it no run on my pc for some reason)

DenFlyvendeGed committed:
Engine store functions implemented

DenFlyvendeGed committed:
added function comments

DenFlyvendeGed committed:
Updated syntax

tr0ub1eM4k3r committed:
Merge branch '80-Create-class-for-engines-group5' of https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web into 81-feature-frontend-for-engine-configuration-group5

DenFlyvendeGed committed:
updated getEngine

DenFlyvendeGed committed:
fixed overflow

DenFlyvendeGed committed:
added tests

DenFlyvendeGed committed:
created more test cases

DenFlyvendeGed committed:
Merge branch 'main' of https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web into 80-Create-class-for-engines-group5

DenFlyvendeGed committed:
Formated

DenFlyvendeGed committed:
Cleanup

tr0ub1eM4k3r committed:
Added adding and deleting engines from the view

tr0ub1eM4k3r committed:
Merge branch '80-Create-class-for-engines-group5' of https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web into 81-feature-frontend-for-engine-configuration-group5

tr0ub1eM4k3r committed:
Made DTO for engines

tr0ub1eM4k3r committed:
Merge branch '81-feature-frontend-for-engine-configuration-group5' of https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web into 81-feature-frontend-for-engine-configuration-group5

tr0ub1eM4k3r committed:
Made engineDTO work

Co-authored-by: SolarEarth37 <SolarEarth37@users.noreply.github.com>

tr0ub1eM4k3r committed:
Added functionality for saving engines in the ui

Co-authored-by: SolarEarth37 <SolarEarth37@users.noreply.github.com>

SolarEarth37 committed:
Made general css for the Engine UI

Co-authored-by: tr0ub1eM4k3r <tr0ub1eM4k3r@users.noreply.github.com>

SolarEarth37 committed:
Made CSS for engine delete button and UI

SolarEarth37 committed:
Added Google icons to deletion UI
Added min-width to avoid panel collapse when no engines

tr0ub1eM4k3r committed:
lint fixes

tr0ub1eM4k3r committed:
Merge branch '81-feature-frontend-for-engine-configuration-group5' of https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web into 81-feature-frontend-for-engine-configuration-group5

tr0ub1eM4k3r committed:
lint changes

SolarEarth37 committed:
Fixed delete UI buttons

SolarEarth37 committed:
Merge branches '81-feature-frontend-for-engine-configuration-group5' and '81-feature-frontend-for-engine-configuration-group5' of https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web into 81-feature-frontend-for-engine-configuration-group5

tr0ub1eM4k3r committed:
Fixed issue where deleted engine will get added

tr0ub1eM4k3r committed:
Made you unable to delete the last engine

tr0ub1eM4k3r committed:
made dialog backdrop

tr0ub1eM4k3r committed:
Merge branch 'main' of
 https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web
into 81-feature-frontend-for-engine-configuration-group5

tr0ub1eM4k3r committed:
Made engineUI openable from the topbar

SolarEarth37 committed:
Incorporated global CSS in Engine UI

DenFlyvendeGed committed:
typo

tr0ub1eM4k3r committed:
Merge branch '80-Create-class-for-engines-group5' of https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web into 81-feature-frontend-for-engine-configuration-group5

tr0ub1eM4k3r committed:
Fixed lint issues

tr0ub1eM4k3r committed:
Merge branch 'main' of https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web into 81-feature-frontend-for-engine-configuration-group5

tr0ub1eM4k3r committed:
Removed unnecessary code

tr0ub1eM4k3r committed:
Changed file names to be more style guide fitting

tr0ub1eM4k3r committed:
Made ports min and max

tr0ub1eM4k3r committed:
Added comments for functions

tr0ub1eM4k3r committed:
Changed naming of interface

tr0ub1eM4k3r committed:
Changed spelling

tr0ub1eM4k3r committed:
Added increased usability when deleing engnies

tr0ub1eM4k3r committed:
Added issue with deleting the last engine

tr0ub1eM4k3r committed:
Merge branch 'main' of https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web
into 81-feature-frontend-for-engine-configuration-group5

tr0ub1eM4k3r committed:
lint

tr0ub1eM4k3r committed:
Rename iEngineStorageObject.ts to IEngineStorageObject.ts

tr0ub1eM4k3r committed:
Rename iEngineUIComponent.ts to IEngineUIComponent.ts

tr0ub1eM4k3r committed:
Rename iModalComponent.ts to IModalComponent.ts

BaBrixx reviewed (changes requested):
## Have not yet looked at code (Will wait till the following changes have been fixed)

## Required changes:
### **Saving incorrect settings:** 
When pressing save while having incorrect settings it simply discards the engine. It should not discard the settings without warning the user about the unsaved changes.

_What should actually happen:_
In the beginning there should be popup warning the user. When the user dismisses this warning they should be returned to the engines modal, with the incorrect input fields being highlighted.

### **General Styling:**
1. Design of the Engine modal
The engine view looks nothing like the one from the old view and generally needs styling. See [this picture](https://raw.githubusercontent.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/main/presentation/EngineConfiguration.png) as a reference.

2. Button Positioning
The position of the buttons should be changed. The delete button besides the name field does not indicate that it deletes the whole engine.
The button could be besides the engine like in the following sketch:
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/765211de-2673-450b-b127-7f95e5ab6f49)

The positioning of the "close" button should be located in the top-right.

### Close functionality
One expects the modal to close whenever one presses outside it unless e.x. one has unsaved changes.

### Local button
The local button from the old GUI is missing again see [this picture](https://raw.githubusercontent.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/main/presentation/EngineConfiguration.png) as a reference.

When toggling the local button the ip-address field should change to a file-path field. In addition to changing the input field validation a new button should also be added to the right of the input field opening a window with the specified path. 

See picture of old GUI below (Path vs. IP):
![image-1](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/f325449e-34eb-4356-b2d8-ce8994e4b8f5)


### Delete engine popup
Conditional for deleting nameless engines, so it says something like `Would you like to delete this engine?` Instead of `the engine:`
![enginedelete](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/c9fb7a33-fb9e-4791-be56-c4fcd769d0b4)

### Options for different engines
We are wondering if the radio buttons for Reveaal, JEcdar and API are necessary, but if they are, maybe a select dropdown menu instead like in the query panel? Not a requirement, just an idea.

Again, this does not exist in the old GUI see [this picture](https://raw.githubusercontent.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/main/presentation/EngineConfiguration.png) as a reference.

tr0ub1eM4k3r committed:
Fixed no prompt when closing with unsaved data
Deleted radio buttons
Css on engineUI bettered

Co-authored-by: SolarEarth37 <SolarEarth37@users.noreply.github.com>

tr0ub1eM4k3r committed:
Merge branch '81-feature-frontend-for-engine-configuration-group5' of https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web into 81-feature-frontend-for-engine-configuration-group5

tr0ub1eM4k3r committed:
Added highlighting of misinformation

Co-authored-by: SolarEarth37 <SolarEarth37@users.noreply.github.com>

tr0ub1eM4k3r committed:
Added interface for the IEngineSeperateComponent

tr0ub1eM4k3r committed:
Merge branch 'main' of https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web into 81-feature-frontend-for-engine-configuration-group5

tr0ub1eM4k3r committed:
Changed static implementation of the EngineStorage

tr0ub1eM4k3r committed:
lint

tr0ub1eM4k3r committed:
Added functionality for closing Modal
when clicking the backdrop

tr0ub1eM4k3r committed:
Set ModalCloseOnBackdrop to false if not defined

tr0ub1eM4k3r committed:
Changed Button component to be more flexible
Implemented the UI for the "use bundle" button

Co-authored-by: SolarEarth37 <SolarEarth37@users.noreply.github.com>

tr0ub1eM4k3r committed:
Merged main

tr0ub1eM4k3r committed:
Moved tabindex

tr0ub1eM4k3r committed:
Moved validation helper functions to own file

DenFlyvendeGed committed:
Merge branch 'main' into 81-feature-frontend-for-engine-configuration-group5

tr0ub1eM4k3r committed:
Merge branch 'main' of https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web into 81-feature-frontend-for-engine-configuration-group5

BaBrixx commented:
You should get a person from your group that has not personally been involved in the development of this PR to do a review before me.

DenFlyvendeGed commented on src/lib/classes/engine/Validation.ts:
```diff
@@ -0,0 +1,24 @@
+const regexTest: RegExp = new RegExp(
+	"^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?).(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?).(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?).(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$",
```
comment with what id does

DenFlyvendeGed commented on src/lib/components/overlayMenu/elements/Button.svelte:
```diff
@@ -6,17 +6,23 @@
 
 	export let text: string;
+
+	export let backgroundColor: string = "white";
```
might clash depending on css theme?

tr0ub1eM4k3r replied:
The component was built to be white by default before, so when changing it, it was set to white by default again to not break the sidepanels and topbar

DenFlyvendeGed commented on yarn-error.log:
should be deleted

DenFlyvendeGed reviewed (commented):
I think it is fine to pull, it has some bugs that should be made issues for though:

sometimes the first engine created seems to triple when you reacces the panel.

when adding and removing valid and unvalid engines in an unknown order, the save functionality either stops working with exeptions printed in the console, or it says that the engines are invalid even though all invalid engines have been deleted.

tr0ub1eM4k3r committed:
Delete yarn-error.log

tr0ub1eM4k3r committed:
Fixed issues with useBundle implementation
Made comment on Regex string in validation

tr0ub1eM4k3r committed:
Merge branch '81-feature-frontend-for-engine-configuration-group5' of https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web into 81-feature-frontend-for-engine-configuration-group5

tr0ub1eM4k3r committed:
Merge branch 'main' into 81-feature-frontend-for-engine-configuration-group5

tr0ub1eM4k3r committed:
Changed color to css variable

SolarEarth37 committed:
Box size changes due to font change

tr0ub1eM4k3r committed:
Fixed issue with error screen on engineUI

tr0ub1eM4k3r committed:
Changed deletion of engines

tr0ub1eM4k3r committed:
Made playwright tests

tr0ub1eM4k3r committed:
Made playwright tests

tr0ub1eM4k3r committed:
lint

tr0ub1eM4k3r committed:
Changed modal css to own variable

DenFlyvendeGed reviewed (dismissed):
LGTM

atjn commented on src/lib/classes/engine/Engine.ts:
```diff
@@ -0,0 +1,129 @@
+		return this.#name;
+	}
+	set name(setName: string | undefined) {
+		if (validateName(setName) && setName !== undefined)
```
```suggestion
		if (validateName(setName))
```
This check is already performed in `validateName`

atjn commented on src/lib/components/engineUI/EngineSeperate.svelte:
```diff
@@ -0,0 +1,372 @@
+					backgroundColor={"var(--engine-ui-button-color)"}
+					buttonColor={"var(--engine-ui-text-color)"}
+				/>
+			{/if}
```
This should not be a button, it should be a check box.

atjn commented on src/lib/components/engineUI/EngineSeperate.svelte:
```diff
@@ -0,0 +1,372 @@
+		<input
+			type="number"
+			placeholder="7000"
+			id="STARTPORT"
```
Ids should always be kebab-case according to our style guide.

atjn commented on src/lib/classes/engine/Engine.ts:
```diff
@@ -0,0 +1,129 @@
+	get type(): EngineType {
+		return this.#type;
+	}
+	set type(engineType: EngineType) {
```
AFAICT there is no engine type?

atjn commented on src/lib/classes/engine/EngineStorage.ts:
```diff
@@ -0,0 +1,165 @@
+		} else {
+			throw new Error("Engine Id does not exist");
+		}
+	}
```
Why is a function like this so complicated? If all engines have numeric IDs, put the engine in the array at the index that corresponds to their ID. Then when you want to delete the engine, just set that index to `undefined`.

atjn commented on src/lib/components/dialogPopover/Modal.svelte:
```diff
@@ -0,0 +1,62 @@
+<script lang="ts">
```
Modals, dialogs and popovers are three separate things. I would assume that this is a modal/modal.svelte

atjn commented on src/lib/classes/engine/Validation.ts:
```diff
@@ -0,0 +1,24 @@
+const regexTest: RegExp = new RegExp(
+	"^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?).(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?).(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?).(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$", // Will test for any valid IP address
```
Why not use [`URL.canParse`](https://developer.mozilla.org/en-US/docs/Web/API/URL/canParse_static) for this? That would also allow you to support urls and things like `localhost`. I see no reason not to support that.

For the input in the DOM, you can just say `<input type="url">` and you're done.

atjn replied:
still here

DenFlyvendeGed replied:
Jup, URL.canParse er ikke fuldt implementeret i Oracle hvilket er grunden til at det ikke blev ændret

atjn replied:
Jup, men I skal stadig acceptere hvis brugeren skriver en URL adresse ind, så I bliver nødt til at gøre et eller andet. Der er mange måder at løse dette på, en af dem er at læse sig frem til at Opera begynder at understøtte det i Januar 2024, så det har nok været understøttet rigeligt længe når UI'et begynder at blive brugt om ~2 år.

atjn commented on src/lib/components/engineUI/EngineSeperate.svelte:
```diff
@@ -0,0 +1,372 @@
+			style="--borderColour: {ipBorderColour}; color: {currentEngine.useBundle
+				? 'grey'
+				: 'var(--engine-ui-text-color)'}"
+			readonly={currentEngine.useBundle}
```
Vil tro det er bedre at gøre det til et disabled input når en brundled engine bliver brugt

BaBrixx commented on src/lib/components/engineUI/EngineSeperate.svelte:
```diff
@@ -0,0 +1,372 @@
+		validateStartPort,
+	} from "$lib/classes/engine/Validation";
+
+	let formElement: HTMLFormElement;
+	let modalContainer: Modal & iModalComponent;
+	let nameContainer: HTMLInputElement;
+	let ipAddressContainer: HTMLInputElement;
+	let startPortContainer: HTMLInputElement;
+	let endPortContainer: HTMLInputElement;
```
Instead of binding to HTML elements you should use the bind:value. This would simplify the code quite a lot!

BaBrixx commented on src/lib/components/engineUI/EngineSeperate.svelte:
The text of input fields should be wrapped in some sort of element e.g. `<p>text</p>`

BaBrixx reviewed (changes requested):
## Non-code Comments
1. The buttons should be fixed:
Padding on the buttons is wrong (See below pictures aswell):
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/316217a6-57da-4a0d-8f61-1de9a78dc350)

2. Alignment doesn't work
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/1637a163-fbbb-49b7-803f-cb41d70a6178)

3. Is not centered (Uses hardcoded padding on all sides)
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/175db259-a2c1-49f0-b1ec-a09f293161b2)
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/88c647a5-9cc7-43b4-8e7a-c7b46076f2b6)
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/ff581122-22fb-45ce-bdab-fd079046f25c)



4. Tabbing does not function as expected
When focusing on the save button the one has to press tab 4 times to tab to the next button. The next button is the close button instead of there add button. 

When pressing tab after having the following button selected it disappears followed by it selecting the button to the right after 3 more tabs:
<img width="42" alt="image" src="https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/ab73f6f1-44a5-4a37-8d0b-6fb14b7ac3b2">
<img width="63" alt="image" src="https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/b6cc642b-693f-4419-860d-6ce64d58d8e1">


5. Bundle
When pressing use bundle there should be better visual clues that an ip address is not needed such as completely removing the input field.

6. Weird background on "Engines":
<img width="124" alt="image" src="https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/a7cfed27-4984-43f4-8c37-32c3d0ce8ae0">

7. Escape
Pressing escape should result in the same popup as pressing X

8. Different appearance in Firefox and chrome (Text)
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/61fa5c60-788a-42c5-9866-353790b6d574)

9. Port input fields allow decimals
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/0b3f39aa-045c-49c7-b03a-d32295d228fd)

10. Closing modal unexpected functionality
Hvis man lukker modal med escape efter at have slettet alle engines og genåbner den så tilføjer den ikke automatisk en model som den ellers ville gøre hvis man gemte inden man lukkede den.

11. Unsaved changes popup
Selvom at man ikke har ændret i noget når man åbner for første gang siger den at man har lavet ændringer

12. At some random times the engines are duplicated
Unable to find a stable way of triggering this bug
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/d24d9273-f5e4-48a3-951d-919e6037fa7d)

14. Valid information alerting about the input not being valid.
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/6c2f0ed6-a3e6-49eb-a52c-80cb89d4543f)


16. It is possible to have 0 engines:
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/6b5eb129-7e23-42dc-92c7-f6e675b853fe)
Add 1 engine, delete top engine, delete only remaining engine: No engines remaining

17. You should store the engine configuration using indexedDB and localStorage

Other weird behaviour:
Deleting engins followed by discarding changes results in the engines not showing up:
![](https://i.gyazo.com/454e9c560debdd03152f7cd2d317cef0.gif)


## Code related:
1. Please refrain from using inline styling if possible

2. Nesting forms in is illegal:
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/6a49e5f7-1d4a-4252-aa37-6502381cad98)
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/5e868b1e-d35f-4e32-9aa7-17eaf9336354)
Instead of forms consider using a div with a button.

tr0ub1eM4k3r committed:
Removed nested forms
Changed to use of SvgButtons ( not done)
changed incorrect spelling
Changed to use of kebab-case for ids
Changed Engine and EngineStorage

SolarEarth37 commented on src/lib/components/engineUI/EngineSeperate.svelte:
```diff
@@ -0,0 +1,372 @@
+			value={currentEngine.name}
+			on:change={onNameChange}
+			bind:this={nameContainer}
+			style="--borderColour: {nameBorderColour}"
```
It's a variable for changing underline color on error

DenFlyvendeGed committed:
fixed tests

SolarEarth37 commented on src/lib/components/engineUI/EngineSeperate.svelte:
```diff
@@ -0,0 +1,372 @@
+	}
+
+	.port-input {
+		width: 37%;
```
An issue has been created for this

SolarEarth37 committed:
Fixed multiple issues, related to comments

DenFlyvendeGed committed:
engineArray uses id as index

DenFlyvendeGed commented on src/lib/classes/engine/EngineStorage.ts:
```diff
@@ -0,0 +1,165 @@
+		index = this.engineArray.findIndex((engine: Engine) => {
+			return engine.id === engine.id;
+		});
+		if (engine === undefined || index > -1) this.#defaultEngine = engine;
```
Incase the default engine gets deleted

DenFlyvendeGed committed:
Removed undefined elements from getEngineArray

SolarEarth37 committed:
Fixed problem with hard coded EngineUI

SolarEarth37 committed:
lint and changes from button to SvgButton

atjn commented on src/lib/components/modal/Modal.svelte:
```diff
@@ -0,0 +1,62 @@
+			dialogContainer.close();
+		}
+	}
+</script>
```
Det ville være endnu bedre hvis close og open var implementeret som en svelte store, da de interagerer meget godt med sveltes native APIer. Jeg har lavet det samme i popover menuen, kig evt. på det. Ellers ser det godt ud.

atjn commented on src/lib/components/engineUI/EngineSeperate.svelte:
```diff
@@ -0,0 +1,391 @@
+			closeModal();
+			return;
+		}
+		currentComponent?.$destroy();
```
It should not be necessary to use JS calls for this. If you implemented the components to react when you change the array, Svelte will automatically destroy this for you.

BaBrixx reviewed (dismissed):
The following points from the previous review are still problems: 
- 7, 9, 10, 11, 14, 15 and other wierd behaviour

We also found the following:
- **_Bundle error_**
You cannot save changes to engines that have previously been saved with the ”Use Bundle checked”. Steps: Save an engine with Use Bundle checked, Open the UI, make a change (except for toggling bundle), and try to save.
- **_TypeErrors in console:_**
setUpEngineSeperate when i attempt to save after deleting an engine (specifically number two of three)
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/435c687a-04eb-4852-a3ed-97f373bda259)

closeModal is not a function when deleting:
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/51baf33b-fd80-4ac3-bc31-1f6fe4e0b86a)

DenFlyvendeGed committed:
fixed port input accepting decimals

SolarEarth37 committed:
Fixed Bundle problem, along with closeModal not working as expected

SolarEarth37 committed:
Merge branch '81-feature-frontend-for-engine-configuration-group5' of https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web into 81-feature-frontend-for-engine-configuration-group5

DenFlyvendeGed committed:
should have fixed 9 -13. tests incoming

DenFlyvendeGed committed:
fixed ip input not centering custom value

DenFlyvendeGed committed:
Use bundle label trigger checkbox. test  are wip

DenFlyvendeGed committed:
added test

DenFlyvendeGed committed:
format

DenFlyvendeGed committed:
Fixed most lint, only one left is label control which might be irelavent?

DenFlyvendeGed committed:
Added engine store to global state, and removed tempengine store from global state

KamyaPA committed:
fixed a11y-label-has-associated-control

KamyaPA committed:
Merge branch '81-feature-frontend-for-engine-configuration-group5' of github.com:ECDAR-AAU-SW-P5/Ecdar-GUI-Web into 81-feature-frontend-for-engine-configuration-group5

DenFlyvendeGed committed:
cleanup

DenFlyvendeGed committed:
lint

DenFlyvendeGed committed:
Fixed being able to delete all engines

DenFlyvendeGed committed:
Removed old functionality for not deleting last element

DenFlyvendeGed committed:
fixed tests

KamyaPA committed:
Added URL.canParse

DenFlyvendeGed committed:
readded ip support

KamyaPA committed:
Merge branch 'main' into 81-feature-frontend-for-engine-configuration-group5

KamyaPA committed:
Refactored to new modal

DenFlyvendeGed committed:
Main merge fixed

KamyaPA committed:
Minor deletions

KamyaPA reviewed (dismissed):
LGTM

KamyaPA commented on src/lib/components/engineUI/EnginePanel.svelte:
```diff
@@ -0,0 +1,66 @@
+			currentEngine={engine}
+			{tempEngines}
+			bind:this={engineSeperateArray[index]}
+		/>
```
This does not break the dynamic update

atjn commented on src/lib/classes/engine/EngineStorage.ts:
```diff
@@ -0,0 +1,162 @@
+	#engineId: number = 0;
+
+	get engineId(): number {
+		return (this.engineId = ++this.#engineId); //auto inc
```
If you are trying to read and understand the code, this can be very difficult to understand. Just increment the private counter on line 19, then return the new value on line 20.

BaBrixx reviewed (dismissed):
### Non code related things
1. If you press escape while having the modal open it is not possible to reopen it.
2. It is possible to save invalid information by simply pressing discard instead of save
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/e79c8a8d-e6fd-44fb-ba53-12257bf542c4)

3. Invalid property value
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/949c8697-77f9-42cc-ae39-641d22c55ad1)

4. Hard to distinguish between different engines (no divider). 
5. Scrollbar needs styling (Becomes invisible when hovering)
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/8e169c0b-3461-43fb-a7f0-2310b525c7c3)

6. Alignment
![image](https://github.com/ECDAR-AAU-SW-P5/Ecdar-GUI-Web/assets/90306530/e4bba8c6-6891-4ff9-b593-8f889f1d9f7a)

DenFlyvendeGed committed:
fixed engine store not being utilized + escape press bug

DenFlyvendeGed commented on src/lib/classes/engine/Validation.ts:
```diff
@@ -0,0 +1,24 @@
+);
+
+export function validateIP(ipAdress: string): boolean {
+	return URL.canParse(ipAdress) || regexTest.test(ipAdress);
```
This seems to accept almost any string, but i trust it works :)

DenFlyvendeGed committed:
Merge remote-tracking branch 'origin/main' into 81-feature-frontend-for-engine-configuration-group5

DenFlyvendeGed commented:
I'm sorry to say, but I'm not very good with the last css :(
so hints would be welcome :)

KamyaPA reviewed (dismissed):
LGTM

SolarEarth37 committed:
Fixed input placeholder color

SolarEarth37 committed:
Fixed scrollbar CSS issues, and light mode being a random combination of colors
